crates/wac-graph/src/plug.rs

```diff
@@ -0,0 +1,64 @@
+use crate::{
+    types::{Package, SubtypeChecker},
+    CompositionGraph, EncodeOptions, NodeId, PackageId,
+};
+
+/// Take the exports of the plug component and plug them into the socket component.
+pub fn plug(
+    graph: &mut CompositionGraph,
+    plugs: Vec<Package>,
+    socket: PackageId,
+    socket_instantiation: NodeId,
+) -> Result<Vec<u8>, anyhow::Error> {
```

Comment:
We don't currently return an `anyhow::Error` from any function in the `wac-graph` crate. It might seem a bit like overkill, but we may want a bespoke error here too. At the very least:

```rust
enum PlugError {
  /// The socket component had no matching imports for the plugs that were provided
  NoPlugHappened,
  /// An error occurred when building the composition graph
  GraphError(anyhow::Error)
}
```